api/src/controllers/users.js

```diff
@@ -9,7 +9,7 @@ const bulkUploadLog = require('../services/bulk-upload-log');
 
 const hasFullPermission = req => {
   return auth
-    .check(req, 'can_update_users')
+    .check(req, ['can_edit', 'can_update_users'])
```

Comment:
I think this should use `hasAllPermissions` not `check`. The difference is that `check` also does some work to see if you're allowed to work in a given part of the hierarchy, but you're not passing in the third parameter so that work is wasted.

```suggestion
    .getUserCtx(req)
    .then(userCtx => auth.hasAllPermissions(userCtx, ['can_edit', 'can_update_users']))
```

Reply:
Yeah the name is a bit generic and looks like this is the preferred function around for some reason.

I checked the usages of `check` function and we're not using the third parameter, maybe it was used in an old code that doesn't exists anymore and we forgot to clean this function. What I like about this function is that it's already extracting the `userCtx` before checking for permissions. 

I've refactored the function to:
- Keep extracting the `userCtx`
- Keep checking for permissions
- Stop checking if user is admin, because this is already done by `hasAllPermissions`
- Stop checking for the facility id, because it's not currently used and we can always add it later if we need it again.
- Return the  `userCtx` if user has all permissions, basically we know what to expect since we talk in `userCtx` schema/definition "terms". Then `userCtx.name` and `userCtx.roles` is available.